components/script/dom/css.rs

```diff
@@ -3,19 +3,42 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 use cssparser::{Parser, ParserInput, serialize_identifier};
+use dom::bindings::codegen::Bindings::DocumentBinding::DocumentBinding::DocumentMethods;
 use dom::bindings::codegen::Bindings::PropertyDescriptorDictBinding::PropertyDescriptorDict;
 use dom::bindings::codegen::Bindings::WindowBinding::WindowBinding::WindowMethods;
 use dom::bindings::error::{Error, Fallible};
+use dom::bindings::inheritance::Castable;
 use dom::bindings::reflector::Reflector;
 use dom::bindings::str::DOMString;
+use dom::node::{Node, NodeDamage};
 use dom::window::Window;
 use dom_struct::dom_struct;
 use style::context::QuirksMode;
 use style::parser::ParserContext;
+use style::properties_and_values::{self, PropertyRegistrationResult};
 use style::stylesheets::CssRuleType;
 use style::stylesheets::supports_rule::{Declaration, parse_condition_or_declaration};
 use style_traits::PARSING_MODE_DEFAULT;
 
+fn handle_property_registration_result(
+    win: &Window,
+    result: PropertyRegistrationResult
+) -> Result<(), Error> {
+    match result {
+        PropertyRegistrationResult::Ok => {
+            // Should lead to the RestyleHint::restyle_subtree() being inserted
+            // eventually in handle_reflow due to checks in Stylist.
```

Comment:
No need to do this. Afaict you could just mark the document's `stylesheet` list as dirty from here, and everything else should follow. You'd even avoid needing the `registered_property_set_generation`!

Reply:
Nice! Would be cool if this kind of thing were documented.

Reply:
Neat, this seems to be another consequence of the spec change.